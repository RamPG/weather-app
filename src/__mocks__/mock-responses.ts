import {
  GetGeoCityNameResponseType,
  GetGeoCoordsResponseType,
  GetWeatherResponseType,
} from '../types/response-types';

export const weatherResponseMock: GetWeatherResponseType = {
  lat: 25.5,
  lon: 24.4,
  timezone: '1200',
  timezone_offset: 3600,
  daily: [
    {
      dt: 5,
      sunrise: 5,
      sunset: 5,
      temp: {
        day: 25,
        min: 20,
        max: 27,
        night: 15,
        eve: 15,
        morn: 15,
      },
      feels_like: {
        day: 25,
        night: 21,
        eve: 23,
        morn: 24,
      },
      pressure: 10,
      humidity: 50,
      dew_point: 1,
      uvi: 1,
      clouds: 1,
      wind_speed: 34,
      wind_deg: 21,
      weather: [
        {
          id: 1,
          main: 'Clouds',
          icon: '04n',
        },
      ],
      pop: 25,
      rain: 15,
    },
  ],
  current: {
    dt: 5,
    sunrise: 5,
    sunset: 5,
    temp: 25,
    feels_like: 23,
    pressure: 10,
    humidity: 50,
    dew_point: 1,
    uvi: 1,
    clouds: 1,
    visibility: 1,
    wind_speed: 34,
    wind_deg: 21,
    weather: [
      {
        id: 1,
        main: 'Clouds',
        icon: '04n',
      },
    ],
  },
};

export const coordsResponseMock: GetGeoCoordsResponseType = {
  place_id: '1',
  licence: '1',
  osm_type: '1',
  osm_id: '1',
  boundingbox: ['1'],
  lat: '25.5',
  lon: '24.4',
  display_name: 'Moscow, Russia',
  class: '1',
  type: '1',
  importance: 1,
  icon: '1',
};

export const cityNameResponseMock: GetGeoCityNameResponseType = {
  place_id: '1',
  licence: '1',
  osm_type: '1',
  osm_id: '1',
  boundingbox: ['1'],
  lat: '25.5',
  lon: '24.4',
  display_name: 'Moscow, Russia',
  address: {
    city: 'Moscow',
    county: 'Moscow District',
    state: '',
    country: 'Russia',
    counry_code: '05',
  },
};
