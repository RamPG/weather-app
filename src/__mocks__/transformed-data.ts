import { DataCoordsStateType, TransformedCurrentDataType, TransformedDailyDataType } from '../types/state-types';

export const transformedDataCoords: DataCoordsStateType = {
  location: 'Moscow',
  latitude: 25.5,
  longitude: 24.4,
};

export const transformedDataCurrent: TransformedCurrentDataType = {
  imgLink: 'https://openweathermap.org/img/wn/04n@2x.png',
  temp: '-249째C',
  feelsLike: '-251째C',
  humidity: 50,
  weather: 'Clouds',
  windSpeed: 34,
};

export const transformedDataDaily: Array<TransformedDailyDataType> = [
  {
    id: 0,
    imgLink: 'https://openweathermap.org/img/wn/04n@2x.png',
    weekDayName: 'Friday',
    monthDay: 14,
    monthDayName: 'August',
    temp: {
      day: '-249째C',
      night: '-259째C',
    },
    weather: 'Clouds',
  },
];
