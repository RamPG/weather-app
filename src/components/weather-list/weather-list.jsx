import React, { useEffect } from 'react';
import { connect } from 'react-redux';

import './weather-list.scss';

import { cityWeatherSevenDaysFetch } from '../../actions/actions';
import withWeatherApi from '../../hoc';
import WeatherListItem from '../weather-list-item';

const WeatherList = ({ cityWeatherSevenDaysFetch, location, daily: { data, loading, error } }) => {
  useEffect(() => {
    cityWeatherSevenDaysFetch();
  }, [location]);
  if (loading) {
    return <h1>Loading...</h1>;
  }
  if (error) {
    return <h1>Error!</h1>;
  }
  return (
    <seciton className="weather-forecast">
      <h1 className="weather-forecast__title">
        Weather for next six days
      </h1>
      <ul className="weather-forecast__weather-list weather-list">
        {
          data.slice(1).map((element) => (
            <li key={element.id}>
              <WeatherListItem
                {...element}
              />
            </li>
          ))
        }
      </ul>
    </seciton>
  );
};
const mapStateToProps = ({ daily, location }) => ({
  daily,
  location,
});
const mapDispatchToProps = (dispatch, { weatherApi }) => ({
  cityWeatherSevenDaysFetch: () => dispatch(cityWeatherSevenDaysFetch(weatherApi)),
});
export default withWeatherApi(connect(mapStateToProps, mapDispatchToProps)(WeatherList));
