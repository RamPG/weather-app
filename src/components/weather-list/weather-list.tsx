import React, { FunctionComponent } from 'react';
import { useSelector } from 'react-redux';

import './weather-list.scss';

import { WeatherListItem } from '../weather-list-item';
import { DataDailyStateType, InitialStateType } from '../../types/state-types';

type WeatherListRenderPropsType = {
    data: Array<DataDailyStateType>,
};

const WeatherListRender: FunctionComponent<WeatherListRenderPropsType> = ({ data }) => (
  <section className="weather-forecast">
    <h1 className="weather-forecast__title">
      Weather for next six days
    </h1>
    <ul className="weather-forecast__weather-list weather-list">
      {
          data.map((element) => (
            <li key={element.id}>
              <WeatherListItem
                {...element}
              />
            </li>
          ))
        }
    </ul>
  </section>
);

export const WeatherList: FunctionComponent = () => {
  const { isLoading, isError, data } = useSelector(({ daily }: InitialStateType) => daily);
  if (isLoading) {
    return <h1>Loading...</h1>;
  }
  if (isError) {
    return <h1>Error!</h1>;
  }
  return <WeatherListRender data={data} />;
};
