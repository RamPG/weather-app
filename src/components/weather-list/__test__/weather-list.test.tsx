import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { WeatherList } from '../weather-list';
import { transformedDataDaily } from '../../../__mocks__/transformed-data';

describe('Weather list component test', () => {
  afterEach(cleanup);
  it('Weather list snapshot', () => {
    const { asFragment } = render(<WeatherList data={transformedDataDaily} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
