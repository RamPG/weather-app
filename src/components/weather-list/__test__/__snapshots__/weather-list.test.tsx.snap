// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather list component test Weather list snapshot 1`] = `
<DocumentFragment>
  <section
    class="weather-forecast"
  >
    <h1
      class="weather-forecast__title"
    >
      Weather for seven days
    </h1>
    <ul
      class="weather-forecast__weather-list weather-list"
    >
      <li>
        <p
          class="weather-list__item-week-day"
        >
          Friday
        </p>
        <p
          class="weather-list__item-date"
        >
          14 August
        </p>
        <p
          class="weather-list__item-img-block"
        >
          <img
            alt="weather"
            class="weather-list__weather-img"
            height="100"
            src="https://openweathermap.org/img/wn/04n@2x.png"
            width="100"
          />
        </p>
        <p
          class="weather-list__item-temp-day"
        >
          -249°C
        </p>
        <p
          class="weather-list__item-temp-night"
        >
          -259°C
        </p>
        <p
          class="weather-list__item-weather"
        >
          Clouds
        </p>
      </li>
    </ul>
  </section>
</DocumentFragment>
`;
